### PR TITLE
[bootstrap] Fix popper config types

### PR DESCRIPTION
**Pull request by kyletsang**
Fixes issues discussed in https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/67333

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://getbootstrap.com/docs/5.3/components/dropdowns/#using-function-with-popperconfig
https://getbootstrap.com/docs/5.3/components/popovers/#using-function-with-popperconfig
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by martin-badin** [comment]
Looks good but please update tests.

File: types/bootstrap/js/dist/popover.d.ts [hunk 1]
```diff
@@ -122,7 +123,7 @@ declare namespace Popover {
         inserted = "inserted.bs.popover",
     }
 
-    type PopperConfigFunction = (defaultBsPopperConfig: Options) => Partial<Options>;
+    type PopperConfigFunction = (defaultBsPopperConfig: Popper.Options) => Partial<Popper.Options>;
 
     interface Options extends Omit<Tooltip.Options, "popperConfig"> {
```
Comment (martin-badin) [inline types/bootstrap/js/dist/popover.d.ts:128]: please remove `Omit`
```suggestion
    interface Options extends Tooltip.Options {
```
and remove `popperConfig` and `PopperConfigFunction` from this file

File: types/bootstrap/js/dist/dropdown.d.ts [hunk 2]
```diff
@@ -78,7 +78,7 @@ declare namespace Dropdown {
 
     type OffsetFunction = () => Offset;
 
-    type PopperConfigFunction = () => Partial<Popper.Options>;
+    type PopperConfigFunction = (defaultBsPopperConfig: Partial<Popper.Options>) => Partial<Popper.Options>;
 
     interface Options {
```
Comment (martin-badin) [inline types/bootstrap/js/dist/dropdown.d.ts:83]: Here please use 
```suggestion
    interface Options extends Pick<Tooltip.Options, "popperConfig">
```

and remove `popperConfig` and `PopperConfigFunction` from this file

**Review by martin-badin — CHANGES_REQUESTED** [review]

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/bootstrap/test/dropdown-tests.ts
1. types/bootstrap/test/popover-tests.ts
1. types/bootstrap/test/tooltip-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 134df016dc37ee9f8225fad224893c74dfe347db
</p>

**Comment by kyletsang** [comment]
Thanks for the review @martin-badin.  I've added the requested changes.

**Review by martin-badin — APPROVED** [review]
Looks good. Thanks

**Comment by luc122c** [comment]
Ready to merge

File: types/bootstrap/js/dist/dropdown.d.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import * as Popper from "@popperjs/core";
 import BaseComponent, { GetInstanceFactory, GetOrCreateInstanceFactory } from "./base-component";
+import Tooltip from "./tooltip";
```
Comment (peterblazejewicz) [inline types/bootstrap/js/dist/dropdown.d.ts:3]: how this can depends on tooltip module? I assume one could extract common type out of tooltip for reuse

**Pull request merged**